firewood/src/db.rs

```diff
@@ -436,13 +436,23 @@ pub struct Db {
 impl Db {
     const PARAM_SIZE: u64 = size_of::<DbParams>() as u64;
 
-    /// Open a database.
-    pub fn new<P: AsRef<Path>>(db_path: P, cfg: &DbConfig) -> Result<Self, DbError> {
-        // TODO: make sure all fds are released at the end
+    pub async fn new<P: AsRef<Path> + Send + Sync + 'static>(
+        db_path: P,
+        cfg: &DbConfig,
+    ) -> Result<Self, api::Error> {
         if cfg.truncate {
-            let _ = std::fs::remove_dir_all(db_path.as_ref());
+            let _ = tokio::fs::remove_dir_all(db_path.as_ref()).await;
         }
 
+        block_in_place(|| Db::new_internal(db_path, cfg.clone()))
+            .map_err(|e| api::Error::InternalError(e.into()))
+    }
+
+    /// Open a database.
+    fn new_internal<P: AsRef<Path> + Send + Sync + 'static>(
```

Comment:
This is over-constrained
```suggestion
    fn new_internal<P: AsRef<Path>>(
```